AppHandling/Run-AlPipeline.ps1

```diff
@@ -1746,11 +1767,36 @@ Write-Host -ForegroundColor Yellow @'
         }
     }
 
+    # Script block for compiling an app using a compiler folder
+    $CompileAppWithBCCompilerFolder = {
+        Param(
+            [scriptblock] $PreCompileApp,
+            [scriptblock] $PostCompileApp,
+            $Parameters
+        )
+
+        if($PreCompileApp) {
+            Write-Host "Running custom pre-compilation script..."
+
+            Invoke-Command -ScriptBlock $PreCompileApp -ArgumentList $Parameters

```

Comment:
I don't think we should transfer all parameters to precompile - only the ones we need
e..g the PostCompile should probably also have the appfile as a parameter.

Reply:
`CompileAppInBcContainer` takes all parameters as input. What are your concerns?

Reply:
Now pre and post compile scripts are called no matter if `useCompilerFolder` is on or off.

For now, we pass all the compilation parameters to both scripts (even as a _ref_ to the pre compile script) + the .app file path to the post compile script.